lnwallet/rpcwallet/sign_coordinator.go

```diff
@@ -0,0 +1,913 @@
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) MuSig2Sign(_ context.Context,
+	in *signrpc.MuSig2SignRequest,
+	_ ...grpc.CallOption) (*signrpc.MuSig2SignResponse, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_MuSig2SignRequest{
+		MuSig2SignRequest: in,
+	}
+
+	return processRequest(
+		s, 0,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) (*signrpc.MuSig2SignResponse,
+			error) {
+
+			rpcResp := resp.GetMuSig2SignResponse()
+			if rpcResp == nil {
+				return nil, ErrUnexpectedResponse
+			}
+
+			return rpcResp, nil
+		},
+	)
+}
+
+// SignMessage sends a SignMessageReq to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignMessage(_ context.Context,
+	in *signrpc.SignMessageReq,
+	_ ...grpc.CallOption) (*signrpc.SignMessageResp, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignMessageReq{
+		SignMessageReq: in,
+	}
+
+	return processRequest(
+		s, 0,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) (*signrpc.SignMessageResp, error) {
+			signResp := resp.GetSignMessageResp()
+			if signResp == nil {
+				return nil, ErrUnexpectedResponse
+			}
+
+			return signResp, nil
+		},
+	)
+}
+
+// SignPsbt sends a SignPsbtRequest to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignPsbt(_ context.Context,
+	in *walletrpc.SignPsbtRequest,
+	_ ...grpc.CallOption) (*walletrpc.SignPsbtResponse, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignPsbtRequest{
+		SignPsbtRequest: in,
+	}
+
+	return processRequest(
+		s, 0,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) (*walletrpc.SignPsbtResponse,
+			error) {
+
+			signResp := resp.GetSignPsbtResponse()
+			if signResp == nil {
+				return nil, ErrUnexpectedResponse
+			}
+
+			return signResp, nil
+		},
+	)
+}
+
+// processRequest is a generic function that sends a request to the remote
+// signer and waits for the corresponding response. If a timeout is set, the
+// function will limit the execution time of the entire function to the
+// specified timeout. If it is not set, configured timeouts will be used for
+// the individual operations within the function.
+func processRequest[R any](
+	s *SignCoordinator,
+	timeout time.Duration,
+	generateRequest func(uint64) walletrpc.SignCoordinatorRequest,
+	extractResponse func(*signerResponse) (R, error),
+) (R, error) {
```

Comment:
nit: fits on previous line... This formatting pattern isn't very nice IMO, with the closing parenthesis unindented.